runtime/lua/vim/lsp/buf.lua

```diff
@@ -422,6 +422,21 @@ function M.clear_references()
   util.buf_clear_references()
 end
 
+--- Requests code actions from all clients and calls the handler exactly once
+--- with all aggregated results
+--@private
+local function code_action_request(params)
```

Comment:
One limitation with this approach compared to formatting is we don't optionally allow a list of client_ids to restrict code actions to a single client, not sure if we should do that though. It may be better with the simpler way as it is.

Reply:
I think for code actions, you probably always want to see the full list. Formatting is different, since there you typically only want one formatter. So probably fine as is?